virtualcluster/pkg/syncer/vnode/vnode.go

```diff
@@ -104,7 +85,31 @@ func NewVirtualNode(provider provider.VirtualNodeProvider, node *corev1.Node) (v
 	return n, nil
 }
 
-var wellKnownNodeLabelsMap = map[string]struct{}{
+// BuildVNodeTaints is used to convert pNode taints to vNode (adding corev1.TaintNodeUnschedulable)
+func BuildVNodeTaints(node *corev1.Node, now metav1.Time) (vnodeTaints []corev1.Taint) {
+	nodeTaints := node.Spec.Taints
```

Comment:
I am not sure if blindly exposing all taints of super node to virtual cluster has any side effects about breaking isolation requirements. Hence, I would suggest having this capability guarded by a feature flag.

Reply:
Would a new field in the VirtualCluster CR that allows a (or excludes) a list of taints like `opaqueMetaPrefixes` & `transparentMetaPrefixes` be useful?

Reply:
As I asked, it depends on how you plan to use the taints. How would the cluster admin know the taints to be included/excluded. For this PR, node taints are blindly copied. Also remember that this field should be fixed during VC lifecycle, is it applicable for taints.

Reply:
> Would a new field in the VirtualCluster CR that allows a (or excludes) a list of taints like `opaqueMetaPrefixes` & `transparentMetaPrefixes` be useful?

I could do the same flag as for labels, but for the list of Taint keys. Initially, I did not think we could expose any isolation or privacy by just seeing the list of taints (it is still the pod syncer and scheduler problems to allow or forbid tolerations  and nodeSelector usage in pods), but yes, let's do it as an another allowlist

Reply:
Added a flag, please take a look

Reply:
@Fei-Guo do you think the PR is good now?